backend/src/service/plants.rs

```diff
@@ -22,9 +22,15 @@ pub async fn find(
     page_parameters: PageParameters,
     pool: &Data<Pool>,
 ) -> Result<Page<PlantsSummaryDto>, ServiceError> {
+    // Empty search queries should be treated like nonexistent queries.
+    let mut search_query: Option<String> = search_parameters.name;
+    if search_query == Some("".to_string()) {
+        search_query = None;
+    }
+
     let mut conn = pool.get().await?;
-    let result = match search_parameters.name {
-        Some(search_query) => Plants::search(&search_query, page_parameters, &mut conn).await?,
+    let result = match search_query {
+        Some(query) => Plants::search(&query, page_parameters, &mut conn).await?,
         None => Plants::find_any(page_parameters, &mut conn).await?,
     };
     Ok(result)
```

Comment:
You can do this like the following instead:
```rust
    let mut conn = pool.get().await?;
    let result = match search_parameters.name {
        Some(query) if !query.is_empty() => {
            Plants::search(&query, page_parameters, &mut conn).await?
        }
        _ => Plants::find_any(page_parameters, &mut conn).await?,
    };
    Ok(result)
```

That uses a guard (see the if inside the match statement) and a default otherwise.